test/test_torch.py

```diff
@@ -11278,6 +11322,30 @@ def test_vander_types(self, device, dtype):
                 atol=1e-3,
                 exact_dtype=False)
 
+    @unittest.skipIf(not TEST_NUMPY, 'NumPy not found')
+    @onlyCPU
+    @dtypes(torch.cfloat, torch.cdouble)
+    def test_vander_complex_types(self, device, dtype):
```

Comment:
You don't need a separate test case, I think? What if you added another case to the existing vander test?

```
elif dtype is torch.cfloat or dtype is torch.cdouble:
  x = [[1 + 1j,...
```

Reply:
The reason I seperated it was because, due to `legacy::cumprod`, `cuda` backend doesn't support `complex dtypes`. The existing case checks with `cpu` and `cuda` both.

So instead of having an `if self.device == 'cuda' and dtype == complexdtype`, made a test with `onlyCPU` decorator which I feel is more explicit in saying the same. 

Once `cumprod` supports `cfloat and cdouble` on CUDA, I believe that is when we should merge the test.

Reply:
Sounds good. Would you add a comment before the new test clarifying that?

Reply:
Actually - one caveat. In the new test, instead of only running it on CPU, would you run it on both and assert that the CUDA version of the test throws a runtime error?

```
if self.device_type == 'cuda':
  with self.assertRaises(RuntimeError):
    ...
```

that way if someone makes cumprod work for complex values on CUDA they'll know to update this test.

Reply:
Makes sense. Will do that.